bluetooth: hci: rpmsg: Trigger fatal error in an assertion handler

ryanjh opened the pull request:
Inform kernel when encountering an unrecoverable condition in order to
dump all pending logging messages.

In addition, log_strdup wrapping around file name was missing.

Signed-off-by: Ryan Chu <ryan.chu@nordicsemi.no>

carlescufi commented on samples/bluetooth/hci_rpmsg/src/main.c:
```diff
@@ -234,7 +234,8 @@ static int hci_rpmsg_send(struct net_buf *buf)
 void bt_ctlr_assert_handle(char *file, uint32_t line)
 {
-	LOG_ERR("Controller assert in: %s at %d", file, line);
+	LOG_ERR("Controller assert in: %s at %d", log_strdup(file), line);
```
Wouldn't it be better to use `ASSERT()` here?

joerchan replied:
BT_ASSERT(false); you mean?

ryanjh replied:
Do you agree if I use BT_ASSERT_MSG() to replace these two lines?

ryanjh replied:
Updated: Using BT_ASSERT_MSG and disabling debug message as before.

ryanjh committed:
samples: nrf5340: Trigger fatal error in an assertion handler

Trigger BT_ASSERT with message to inform kernel when encountering an
unrecoverable condition in order to dump all pending logging messages.

In this change, the log level is configured by CONFIG_BT_LOG_LEVEL

Signed-off-by: Ryan Chu <ryan.chu@nordicsemi.no>

ryanjh commented:
Hi, is there a concern to move this PR forward?
